### PR TITLE
annotation: cleanup and add support, procured and site fields

### PR DESCRIPTION
**Pull request by wdpypere**
Cleaned up the documentation and added 3 fields used at UGent

### CODE REVIEW

**Comment by stdweird** [comment]
@ned21 @gombasg any ideas for a good alternative location for the supprot and procuremnet dates? anything in aquilon metadata perhaps?

File: quattor/types/annotation.pan [hunk 1]
```diff
@@ -5,18 +5,39 @@ declaration template quattor/types/annotation;
...
+    @{clock of peripheral}
+    "clock"        ? long
+    @{language of the product}
+    "lang"         ? string
+    @{power in watts}
+    "power"        ? long
+    @{location of the hardware}
+    "location"     ? string
+    @{campus location of the hardware}
+    "campus"         ? string
```
Comment (jrha) [inline quattor/types/annotation.pan]: We already have `/hardware/sysloc/campus`, `structure_annotation` gets attached to a large number of bits and bobs, we don't this here.

See https://github.com/quattor/template-library-core/blob/master/quattor/types/hardware.pan#L170

File: quattor/types/annotation.pan [hunk 2]
```diff
@@ -5,18 +5,39 @@ declaration template quattor/types/annotation;
...
+    @{Date at which the system support runs. DD-MM-YYYY}
+    "support"      ? string with match(SELF, '^[0-3]\d-[0-1]\d-[1-2]\d{3}$')
+    @{Date at witch the system is procured.}
+    "procured"     ? string with match(SELF, '^[0-3]\d-[0-1]\d-[1-2]\d{3}$')
```
Comment (jrha) [inline quattor/types/annotation.pan]: How about `type_isodate`?

File: quattor/types/annotation.pan (hunk 1 above)
Comment (wdpypere) [inline quattor/types/annotation.pan]: cool! did not know that. I'll remove campus here.

File: quattor/types/annotation.pan (hunk 2 above)
Comment (wdpypere) [inline quattor/types/annotation.pan]: Also did not know that, I'll fix it.

Comment (jrha) [inline quattor/types/annotation.pan]: If the support and procured dates apply to the machines, perhaps they should hang off `structure_hardware`? If they are more generally useful then the annotations should reflect that.

Comment (wdpypere) [inline quattor/types/annotation.pan]: OK, I'll move them.

**Comment by wdpypere** [comment]
@jrha remarks addressed

**Comment by jrha** [comment]
@hpcugentbot test this please

File: quattor/types/hardware.pan [hunk 3]
```diff
@@ -208,6 +208,10 @@ type structure_hardware = {
     "sysloc"       ? structure_sysloc
     "nodename"     ? string
     "benchmarks" ? structure_benchmark
+    @{Date at which the hardware support runs out.}
+    "support"      ? type_isodate
+    @{Date at witch the hardware is procured.}
```
Comment (jrha) [inline quattor/types/hardware.pan]: witch → which

**Comment by wdpypere** [comment]
@jrha remark fixed

**Comment by jrha** [comment]
@hpcugentbot test this please

**Pull request merged**